runtime/lua/vim/treesitter/query.lua

```diff
@@ -1,18 +1,48 @@
 local api = vim.api
 local language = require('vim.treesitter.language')
 
----@class Query
----@field captures string[] List of captures used in query
----@field info TSQueryInfo Contains used queries, predicates, directives
----@field query userdata Parsed query
+local M = {}
+
+---Parsed query, see |vim.treesitter.query.parse()|
+---
+---@class vim.treesitter.query.Query
```

Comment:
```suggestion
---@class vim.treesitter.Query
```

We can shorten this a bit like we have with `vim.diagnostic`.